x/oracle/keeper/tally_test.go

```diff
@@ -0,0 +1,201 @@
+package keeper_test
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	stakingkeeper "github.com/cosmos/cosmos-sdk/x/staking/keeper"
+
+	"github.com/persistenceOne/persistence-sdk/v2/x/oracle/keeper"
+	"github.com/persistenceOne/persistence-sdk/v2/x/oracle/testutil"
+	"github.com/persistenceOne/persistence-sdk/v2/x/oracle/types"
+)
+
+// TestTally is a test for the tallying logic that calculates the median.
+func (s *KeeperTestSuite) TestTally() {
+	testCases := []tallyTestCase{
+		{
+			Description: "basic test for picking a winner",
+			// Tally 2 votes
+			// rewardBand: 1.000000000000000000
+			// weightedMedian: 1.000000000000000000
+			// standardDeviation: 0.707106781186547525
+			// rewardSpread: 0.707106781186547525
+
+			Ballot: types.ExchangeRateBallot{
+				types.NewVoteForTally(sdk.MustNewDecFromStr("1.0"), "AAA", val(1), 10000),
+				types.NewVoteForTally(sdk.MustNewDecFromStr("2.0"), "AAA", val(2), 10000),
+			},
+			RewardBand: sdk.MustNewDecFromStr("1.0"),
+			ValidatorClaimMap: map[string]types.Claim{
+				val(1).String(): types.NewClaim(10000, 0, 0, val(1)),
```

Comment:
It's just a shortcut for generating addresses for Nth validator, since we need only address, there is no need to create n real validators which addresses will be random.

This particular test also uses the fact that `val(n) < val(n+1)` during sorting. Not just random addresses.

In terms of code clarity, `val[1]` is no prettier than `val(1)` ;)